File: 2022/day04/python/main.py
import sys 

# Checking if the user has entered a valid amount of arguments
if len(sys.argv) != 3:
    print("Usage: python main.py <1|2> <input file>")
    exit()

print("This is going to be day04 in python")

# making sure the user has entered valid arguments
if sys.argv[1] == "1":
    # Part 1
    print("Part 1")
elif sys.argv[1] == "2":
    # Part 2
    print("Part 2")
else:
    print("Invalid argument \n Usage: python main.py <1|2> <input file>")
    exit()

File: 2022/day04/c/main.c
#include <stdio.h>
#include <string.h>

int main(int argc,char *argv[]) {

    // Check for the correct number of arguments
    if (argc != 3) {
        printf("Usage: ./day04 <1|2> <input file>\n");
        return 1;
    }

    if (strcmp(argv[1], "1") == 0) {
        printf("Part 1\n");
    } else if (strcmp(argv[1], "2") == 0) {
        printf("Part 2\n");
    } else {
        printf("Usage: ./day04 <1|2> <input file>\n");
        return 1;
    }

    return 0;
}